frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx

```diff
@@ -97,24 +56,38 @@ export const PaymentRedirect = () => {
   }, [paymentDetails.trnApproved]);
 
   useEffect(() => {
-    if (issuedPermit.current === false) {
-      const permitIdsArray = getPermitIdsArray(permitIds);
+    const ids:string[] = [];
+    getPaymentByTransactionId(transactionId)
+      .then((response) => {
+        response?.applicationDetails?.forEach((application) => {
+          if (application?.applicationId) {
+            ids.push(application.applicationId)
+          }
+        })
+      }).finally(() => {
+        setApplicationIds(ids);
+      })
+
+  }, [transactionId]);
+
+  useEffect(() => {
+    if (applicationIds?.length > 0 && issuedPermit.current === false) {
 
-      if (permitIdsArray.length === 0) {
+      if (applicationIds?.length === 0) {
```

Comment:
Line 74 and Line 76 are contradictory if checks. applicationIds?.length > 0  and applicationIds?.length === 0